src/interface/meshblock_data.cpp

```diff
@@ -234,25 +234,25 @@ std::shared_ptr<MeshBlockData<T>> MeshBlockData<T>::SparseSlice(int id) {
 ///        indices represent inclusive bounds for, e.g., a sparse or tensor-valued
 ///        variable.
 template <typename T>
-VariableFluxPack<T> MeshBlockData<T>::PackVariablesAndFluxesHelper_(
-    const std::vector<std::string> &var_names, const std::vector<std::string> &flx_names,
-    const vpack_types::VarList<T> &vars, const vpack_types::VarList<T> &fvars,
-    PackIndexMap &vmap, vpack_types::StringPair &key) {
-  key = std::make_pair(var_names, flx_names);
-  auto kvpair = varFluxPackMap_.find(key);
+PackVariablesAndFluxesResult<T>
```

Comment:
What's the motivation for returning the map by value instead of by reference? This seems equivalent to me?

Reply:
I think the motivation is to avoid output variables. If I remember correctly, the Google style guide says to never use non-const references, instead to use pointers when a function changes its argument. This function really has 3 outputs: the VariableFluxPack, the PackIndexMap, and the key. One is returned as a return value, the others are set as output variables passed by reference. This is stylistically not very elegant, so Andrew introduced the `PackVariablesAndFluxesResult` type that combines all 3 outputs into one struct that can be returned, thus removing the need for output variables.

Reply:
I don't really agree that returning a massive struct by value is stylistically better than modifying input variables by reference. Personally, I find this uglier, but I'm willing to be over-ruled.

One bigger question though. Which version is more performant?